packages/kit/test/apps/basics/svelte.config.js

```diff
@@ -6,16 +6,27 @@ const config = {
 		adapter: {
 			name: 'test-adapter',
 			adapt() {},
-			emulate() {
+			emulate(opts) {
 				return {
+					async beforeRequest(req, res, next) {
+						const middleware = await opts.importEntryPoint('test-adapter-middleware');
+						await middleware.default(req, res, next);
+					},
 					platform({ config, prerender }) {
 						return { config, prerender };
 					}
 				};
 			},
 			supports: {
 				read: () => true
-			}
+			},
+			additionalEntryPoints: () => [
+				{
+					name: 'test-adapter-middleware',
+					file: 'test-adapter-middleware.js',
+					allowedFeatures: []
+				}
+			]
```

Comment:
Does this need to be a function? Feels like it could just be this:

```suggestion
			additionalEntryPoints: {
				'test-adapter-middleware': {
					file: 'test-adapter-middleware.js',
					allowedFeatures: []
				}
			}
```

If we need to make it parametric in the future then we could, but unless we have an idea of what parameters could be passed here then it might be better to file it under YAGNI

Reply:
In fact I'm very confident that we wouldn't need to make it parametric in future, since the adapter itself is creating this object with knowledge of its options

Reply:
If we use the existing `supports` mechanism, instead of expecting adapter authors to know how to map e.g. `supports.read(...)` to `"$app/server:read"`, then we could make it simpler still

```suggestion
			additionalEntryPoints: {
				'test-adapter-middleware': 'test-adapter-middleware.js'
			}
```

Reply:
> Does this need to be a function?

The idea was that maybe someone needs to do something async on the fly when generating these. But I since just used `fs.existsSync` etc so yeah we can make this just an array if we think there nothing changing here 

> Can we use the existing supports mechanism instead of disallowedFeatures?

That would mean adjusting when and how `supports` is called. Right now it's only called for routes, and I'm not sure how that could be adjusted in a way that makes sense. Like, you either get an object with a route etc on it or just the name of the entry point? Feels a bit messy

Reply:
Yes it would require a change in signature, but that's surely much less messy than what we have right now?